source/code/plugins/agent_maintenance_script.rb

```diff
@@ -0,0 +1,639 @@
+require 'optparse'
+
+module MaintenanceModule
+
+  class Maintenance
+    require 'openssl'
+    require 'fileutils'
+    require 'net/http'
+    require 'uri'
+    require 'gyoku'
+    require 'syslog/logger'
+    require 'etc'
+
+    require_relative 'oms_common'
+    require_relative 'agent_topology_request_script'
+    require_relative 'oms_configuration'
+
+    attr_reader :AGENT_USER, :load_config_success
+    attr_accessor :suppress_output
+  
+    def initialize(omsadmin_conf_path, cert_path, key_path, proxy_path, os_info, install_info, verbose = false)
+      @suppress_output = true
+
+      @AGENT_USER = "omsagent"
+      @AGENT_GROUP = "omiusers"
+      @omsadmin_conf_path = omsadmin_conf_path
+      @cert_path = cert_path
+      @key_path = key_path
+      @proxy_path = proxy_path
+      @os_info = os_info
+      @install_info = install_info
+      @verbose = verbose
+      # Config to be read/written from omsadmin.conf
+      @WORKSPACE_ID = nil
+      @AGENT_GUID = nil
+      @URL_TLD = nil
+      @FQDN = nil
+      @LOG_FACILITY = nil
+      @CERTIFICATE_UPDATE_ENDPOINT = nil
+
+      @load_config_success = load_config
+      init_logger
+
+      @suppress_output = false
+    end
+
+    # Assign logger from log facility in omsadmin.conf
+    def init_logger
+      facility = case @LOG_FACILITY
+          # Custom log facilities supported by both Ruby and bash logger
+          when "auth"     then Syslog::LOG_AUTHPRIV  # LOG_AUTH is deprecated
+          when "authpriv" then Syslog::LOG_AUTHPRIV
+          when "cron"     then Syslog::LOG_CRON
+          when "daemon"   then Syslog::LOG_DAEMON
+          when "ftp"      then Syslog::LOG_FTP
+          when "kern"     then Syslog::LOG_KERN
+          when "lpr"      then Syslog::LOG_LRP
+          when "mail"     then Syslog::LOG_MAIL
+          when "news"     then Syslog::LOG_NEWS
+          when "security" then Syslog::LOG_SECURITY
+          when "syslog"   then Syslog::LOG_SYSLOG
+          when "user"     then Syslog::LOG_USER
+          when "uucp"     then Syslog::LOG_UUCP
+
+          when "local0"   then Syslog::LOG_LOCAL0
+          when "local1"   then Syslog::LOG_LOCAL1
+          when "local2"   then Syslog::LOG_LOCAL2
+          when "local3"   then Syslog::LOG_LOCAL3
+          when "local4"   then Syslog::LOG_LOCAL4
+          when "local5"   then Syslog::LOG_LOCAL5
+          when "local6"   then Syslog::LOG_LOCAL6
+          when "local7"   then Syslog::LOG_LOCAL7
+
+          # default logger will be local0
+          else Syslog::LOG_LOCAL0
+      end
+
+      if !Syslog.opened?
+        Syslog::Logger.syslog = Syslog.open("omsagent", Syslog::LOG_PID, facility)
+      end
+      @logger = Syslog::Logger.new
+    end
+
+    # Return true if the current executing user is root
+    def is_current_user_root
+      return true if (Process.euid == 0)
```

Comment:
During testing I received errors when simple returning a boolean evaluation: "void value expression" and this fixed that.